modules/virt-managing-auto-update-all-system-boot-sources.adoc

```diff
@@ -0,0 +1,39 @@
+// Module included in the following assembly:
+//
+// * virt/vm_templates/virt-automatic-bootsource-updates.adoc
+//
+
+:_content-type: PROCEDURE
+[id="virt-managing-auto-update-all-system-boot-sources_{context}"]
+= Managing automatic updates for all system-defined boot sources
+
+Disabling automatic boot source imports and updates can be helpful to reduce the number of logs in disconnected environments or to reduce resource usage.
+
+To disable automatic updates for all system-defined boot sources, turn off the `enableCommonBootImageImport` feature gate by setting the value to `false`. Setting this value to `true` re-enables the feature gate and turns automatic updates back on.
```

Comment:
It will just stop the DataImportCron updates. DataSources and golden image PVCs won't be touched.
@nunnatsa ack?

Reply:
No. The DataImportCronTemplate will be removed from SSP. I almost 100% sure it will then remove the DataImportCron CRs

Reply:
This is a question to CDI or SSP team.

But in general - we do want everything deleted in this case. e.g. in disconnected environment, pulling the images will fail anyway.

Reply:
Afaik in disconnected env `DVs` still use their `sourceRef` for `DataSource` referring a local `PVC`.
@akrejcir @dominikholler ?

Reply:
When `DataImportCronTemplate` is removed from SSP CR, `DataSources` are reverted to their default value. So they will not point to PVC that was created by `DataImportCron`.

I don't know about disconnected env.

Reply:
@arnongilboa @akrejcir , this is a bit over my head :) can you clarify the impact to the user in this situation? 

The way the introduction is currently phrased makes this seem like a harmless action: "Disabling automatic boot source imports and updates can be helpful to reduce the number of logs in disconnected environments or to reduce resource usage."

If the actual impact of turning it off is disruptive, I want to be really clear about the ways in which the user(s) will experience that disruption. Thanks for your help!

Reply:
In `disconnected environments`, users wil get `CDIDataImportCronOutdated` alerts, as `DataImportCron` cannot poll or import the latest disk image versions. Setting `enableCommonBootImageImport` to `false` will eliminate these alerts so I guess that was meant by `reduce the number of logs in disconnected environments`. Afaiu, this is not considered disruptive, as DataImportCron was not operational anyway in this case. As @akrejcir & @nunnatsa commented, this will remove `DataImportCronTemplate` from SSP CR, and `DataSources` are reverted to their default value.

Reply:
> ...this will remove DataImportCronTemplate from SSP CR, and DataSources are reverted to their default value.

Unless there is a clear impact on the user/admin, I don't know if there is a strong case for adding these details to the downstream docs. We tend to avoid documenting what is happening behind the scenes unless it's necessary. Thoughts? @aglitke @arnongilboa @kgoldbla 

I did just add an update to clarify the part about disconnected environments, though.

Reply:
I just tested what happens if you disable the `enableCommonBootImageImport` feature gate.  All DataImportCron objects are deleted.  The PVCs in the openshift-virtualization-os-images namespace are not removed but the DataSource objects are reconciled to point to nonexisting PVCs.  For example, the rhel8 DataSource will be changed so that it points to a non-existent PVC named 'rhel8' whereas it previously pointed to rhel8-06c76d29d511.

I think this is the behavior we want since the admin can optionally  delete the previously imported os images if they would like.  Boot sources can be manually added by creating the appropriate DV to provide the populated PVC with the base name (eg. rhel8).

Reply:
Maybe "DataSources are reset to PVCs which can be manually populated with operating system images."
and "Optionally, after disabling the feature gate, administrators may delete the previously created PVCs"